### ISSUE TITLE
README: clarify how the messages are sent (how GCM has been replaced)

### ISSUE THREAD

**Issue by tuxayo** [LibreSignal/LibreSignal#17]
I'm exited that you managed to remove the dependency Google Cloud Messaging, kudos for that!
I'm trying understand how it has been replaced.

> uses your phone's data connection (WiFi/3G/4G) to communicate securely

I couldn't find if there is a server or if the messages are transmitted directly between clients?

In the former case, where is the server? Is the source code Free/Libre/Open? What metadata can it knows?

**Comment by mimi89999** [comment]
It's OWS server over WebSocket.

**Comment by JavaJens** [comment]
GCM was never used for message sending.
Messages are delivered using the Open Whispersystems server, the protocol is explained in the Signal repository.

The server source is partially open source, as the source code for phone signaling (aka. RedPhone server) is not available, whereas the Textsecure server is available

**Comment by tuxayo** [comment]
> GCM was never used for message sending.

What was the dependency to GCM for? Push?

**Comment by mimi89999** [comment]
Yes, for push.

**Comment by tuxayo** [comment]
Is there an impact on power consumption?

**Comment by mimi89999** [comment]
I don't think so.

**Comment by mimi89999** [comment]
There should be a small impact on battery life, but I haven't noticed it.

**Comment by JavaJens** [comment]
Obviously there is an impact if you have a long running network connection.
The benefit of GCM is that instead of having one connection for each application, GCM can have one, coordinated connection, thus saving battery.
The question is how much an impact does it have and I think  all reports say that the impact is not very noticeable.

**Comment by mimi89999** [comment]
Usually at the end of the day, using LibreSignal over Wi-Fi only, my battery report says that LibreSignal used 1%.

**Comment by mimi89999** [comment]
For me that's not an impact on power consumption. I compared LS to other free/libre messaging apps and other are much worse.

**Comment by tuxayo** [comment]
Great, does the web socket connects directly to Open Whisper Systems's server to poll for new messages? Or is there and intermediate server?

**Comment by mimi89999** [comment]
It connects directly to OWS servers.

**Comment by tuxayo** [comment]
Great that it works!

Thanks a lot for those details. I would like to sum them up to eventually add them to the README.

I was thinking to append this to the "WebSocket Support" part:

> For push notifications, Google Cloud Messaging has be replaced by WebSocket to directly connect to Open Whisper Systems's server.

What do you think?

**Comment by mimi89999** [comment]
Were exactly would you put this text? You can open a PR to add it...

**Comment by tuxayo** [comment]
> You can open a PR to add it...

I wanted to get the text right before, to avoid doing many commits to rephrase/add stuff.

Here it is: #23

**Comment by tuxayo** [comment]
Thanks for all the info, now I have to convert someone who couldn't use Signal because of the dependency to Google Play Services. :-)

**Comment by tuxayo** [comment]
@JavaJens :

> The question is how much an impact does it have and I think all reports say that the impact is not very noticeable.

I just found [this](https://github.com/JavaJens/TextSecure/issues/22#issuecomment-127400981)

>  TL;DR: It seems that the total battery drain with websockets on this device is about 10%/h and with GCM about 2%/h. This shows that there is an increased amount of battery drain but without further data I would not use these results to quantify the increased drain. 

Are those numbers still relevant or was there some optimization since? Or were they invalidated? Or this not very noticeable?

**Comment by mimi89999** [comment]
I use LS only over Wi-Fi and the battery usage is > 1%/h

**Comment by mimi89999** [comment]
@tuxayo What is the battery consumption of LS per hour on your device?

**Comment by tuxayo** [comment]
I only use it over Wi-Fi and I don't remember having seen it yet on the consumption reports.

What about the usage on mobile data? (that I can't test)

**Comment by mimi89999** [comment]
AFAIK battery usage, when using LS over mobile data is much worse, but I'm only using it over WI-Fi, so I can't tell you much about it.

**Comment by mimi89999** [comment]
https://github.com/JavaJens/TextSecure/issues/22

**Comment by tuxayo** [comment]
> JavaJens#22

My original message was asking if those numbers were still relevant, thanks for confirming this.

**Comment by mimi89999** [comment]
It depends on your network conditions.
